ArduCopter/fence.cpp

```diff
@@ -10,8 +10,12 @@ void Copter::fence_check()
 {
     const uint8_t orig_breaches = fence.get_breaches();
 
+    bool is_in_landing = flightmode->mode_number() == Mode::Number::LAND
```

Comment:
there is a regression there. For RTL and SMART_RTL we are now disabling the fence as soon as the modes engaged, whereas previously we were disable only when the modes enter in the landing phase. We shouldn't forget that RTL and Smart_rtl have navigation phase where the fences should still be one. 
so the RTL submode or the landing_state flag should be checked for this case

Reply:
probably same issue with Plane/Quadplane

Reply:
Thanks, fixed